tephra-hbase-compat-0.96/src/test/java/co/cask/tephra/hbase96/coprocessor/CellSkipFilterTest.java

```diff
@@ -0,0 +1,97 @@
+import org.apache.hadoop.hbase.util.Bytes;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * HBase 0.96 specific test for testing {@link CellSkipFilter}.
+ */
+public class CellSkipFilterTest {
+  protected static final byte[] FAM = new byte[] {'f'};
+  protected static final byte[] COL = new byte[] {'c'};
+
+  @Test
+  public void testSkipFiltering() throws Exception {
+    Filter filter = new CellSkipFilter(new MyFilter(Filter.ReturnCode.INCLUDE));
+    long timestamp = System.currentTimeMillis();
+    assertEquals(Filter.ReturnCode.INCLUDE, filter.filterKeyValue(newKeyValue("row1", "val1", timestamp)));
+    assertEquals(Filter.ReturnCode.INCLUDE, filter.filterKeyValue(newKeyValue("row1", "val1", timestamp - 1)));
+
+    filter = new CellSkipFilter(new MyFilter(Filter.ReturnCode.INCLUDE_AND_NEXT_COL));
+    assertEquals(Filter.ReturnCode.INCLUDE_AND_NEXT_COL, filter.filterKeyValue(newKeyValue("row1", "val1", timestamp)));
+    assertEquals(Filter.ReturnCode.NEXT_COL, filter.filterKeyValue(newKeyValue("row1", "val1", timestamp - 1)));
+
+    filter = new CellSkipFilter(new MyFilter(Filter.ReturnCode.SKIP));
+    assertEquals(Filter.ReturnCode.SKIP, filter.filterKeyValue(newKeyValue("row1", "val1", timestamp)));
+    assertEquals(Filter.ReturnCode.SKIP, filter.filterKeyValue(newKeyValue("row1", "val1", timestamp - 1)));
+
+    filter = new CellSkipFilter(new MyFilter(Filter.ReturnCode.NEXT_COL));
+    assertEquals(Filter.ReturnCode.NEXT_COL, filter.filterKeyValue(newKeyValue("row1", "val1", timestamp)));
+    assertEquals(Filter.ReturnCode.NEXT_COL, filter.filterKeyValue(newKeyValue("row1", "val1", timestamp - 1)));
+
+    filter = new CellSkipFilter(new MyFilter(Filter.ReturnCode.NEXT_ROW));
+    assertEquals(Filter.ReturnCode.NEXT_ROW, filter.filterKeyValue(newKeyValue("row1", "val1", timestamp)));
+    assertEquals(Filter.ReturnCode.NEXT_ROW, filter.filterKeyValue(newKeyValue("row1", "val1", timestamp - 1)));
+
+    filter = new CellSkipFilter(new MyFilter(Filter.ReturnCode.SEEK_NEXT_USING_HINT));
+    assertEquals(Filter.ReturnCode.SEEK_NEXT_USING_HINT, filter.filterKeyValue(newKeyValue("row1", "val1", timestamp)));
+    assertEquals(Filter.ReturnCode.SEEK_NEXT_USING_HINT,
+                 filter.filterKeyValue(newKeyValue("row1", "val1", timestamp - 1)));
+  }
+
+  protected KeyValue newKeyValue(String rowkey, String value, long timestamp) {
+    return new KeyValue(Bytes.toBytes(rowkey), FAM, COL, timestamp, Bytes.toBytes(value));
+  }
+
+  /**
+   * Sample filter for testing. This filter returns the same code with which it is constructed, except for NEXT_COL.
+   * For NEXT_COL, initially it returns NEXT_COL and after that it always return INCLUDE.
+   */
+  class MyFilter extends FilterBase {
+
+    private final ReturnCode code;
+    private int counter = 0;
+
+    public MyFilter(ReturnCode code) {
```

Comment:
Would it be simpler if we take a list of codes instead of just one? MyFilter will then cycle through all codes. Then it might be easier to write different combination of tests.